packages/ts/react-auth/src/ProtectedRoute.tsx

```diff
@@ -0,0 +1,75 @@
+/* -
+ * Copyright (C) 2023 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full
+ * license.
+ */
+import type { ReactNode } from 'react';
+import type { RouteObject } from 'react-router-dom';
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+import { useAuth, type AccessProps } from './useAuth.js';
+
+interface ProtectedRouteProps {
+  redirectPath: string;
+  route: ReactNode;
```

Comment:
```suggestion
  element: ReactNode;
```
Reading the code I was a bit confused how you could cast  a route object into an element, but then saw that this is effectively the route object's element.